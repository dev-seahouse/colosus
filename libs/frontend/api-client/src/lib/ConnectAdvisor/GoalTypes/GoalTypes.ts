import ConnectAdvisorBasePrivateApi from '../_Base/Base';
import type { ConnectTenantDto } from '@bambu/shared';

export interface ConnectAdvisorGetGoalTypesResponseDto {
  goalTypes: ReadonlyArray<ConnectTenantDto.IConnectTenantGoalTypeForTenantDto>;
}

export interface ConnectAdvisorSetGoalTypesRequestDto {
  /* An array of enabled goal type UUIDs sorted by display order */
  goalTypeIds: ReadonlyArray<string>;
}

export class ConnectAdvisorGoalTypesApi extends ConnectAdvisorBasePrivateApi {
  constructor(private readonly apiPath = 'goal-types') {
    super();
  }

  /**
   * Retrieves list of goal types and whether each is of them is enabled/disabled
   * - {@link http://localhost:9000/openapi#/Connect/GetGoalTypes}
   */
  public async getGoalTypes() {
    return await this.axios.get<ConnectAdvisorGetGoalTypesResponseDto>(
      `${this.apiPath}`
    );
  }

  /**
   * Enables or disables a goal type
   * - ${@link http://localhost:9000/openapi#/Connect/SetGoalTypes}
   */

  public async setGoalTypes(req: ConnectAdvisorSetGoalTypesRequestDto) {
    return await this.axios.post<never>(`${this.apiPath}`, req);
  }
}

export default ConnectAdvisorGoalTypesApi;
