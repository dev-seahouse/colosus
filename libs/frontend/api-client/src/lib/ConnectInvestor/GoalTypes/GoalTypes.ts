import ConnectInvestorBaseApi from '../_Base/Base';
import type { ConnectTenantDto } from '@bambu/shared';

export interface ConnectInvestorGetGoalTypesResponseDto {
  goalTypes: ReadonlyArray<ConnectTenantDto.IConnectTenantGoalTypeDto>;
}

export class ConnectInvestorGoalTypesApi extends ConnectInvestorBaseApi {
  constructor(private readonly apiPath = '/goal-types') {
    super();
  }

  public async getInvestorGoalTypes() {
    return await this.axios.get<ConnectInvestorGetGoalTypesResponseDto>(
      this.apiPath
    );
  }
}

export default ConnectInvestorGoalTypesApi;
