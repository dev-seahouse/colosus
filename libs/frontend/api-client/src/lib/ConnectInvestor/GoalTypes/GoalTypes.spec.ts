import { describe } from 'vitest';
import { ConnectInvestorGoalTypesApi } from './GoalTypes';

describe('ConnectInvestorGoalTypesApi', () => {
  const connectInvestorGoalTypesApi = new ConnectInvestorGoalTypesApi();

  describe('getInvestorGoalTypes', () => {
    it('should return correct response', async () => {
      const res = await connectInvestorGoalTypesApi.getInvestorGoalTypes();
      expect(res.data).toMatchSnapshot();
    });
  });
});
