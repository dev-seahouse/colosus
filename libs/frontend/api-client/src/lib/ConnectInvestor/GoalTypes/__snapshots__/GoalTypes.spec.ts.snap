// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectInvestorGoalTypesApi > getInvestorGoalTypes > should return correct response 1`] = `
{
  "goalTypes": [
    {
      "description": "Retire comfortably",
      "id": "8d299530-c622-47cb-b425-82429d7443c0",
      "name": "Retirement",
    },
    {
      "description": "Buy a house",
      "id": "d0f191ec-5ddc-4f18-81c6-38b89ac7b5a6",
      "name": "House",
    },
    {
      "description": "Save for college fees",
      "id": "03c46eb2-8004-4fb9-b400-ec3e9903b0db",
      "name": "Education",
    },
    {
      "description": "Just want to grow my wealth",
      "id": "b4dd784c-1743-4161-8b9e-d6e460b6c6c4",
      "name": "Growing Wealth",
    },
    {
      "description": "I have another goal in mind",
      "id": "a4d7abd5-dcbc-44b7-86ca-a0b0cbbd3465",
      "name": "Other",
    },
  ],
}
`;
